server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1167,6 +1172,16 @@ public VolumeVO resizeVolume(ResizeVolumeCmd cmd) throws ResourceAllocationExcep
                 shrinkOk);
     }
 
+    /**
+     * A volume should not be resized if it covers ALL the following scenarios: <br>
+     * 1 - Root volume <br>
+     * 2 - && Current Disk Offering enforces a root disk size (in this case one can resize only by changing the Service Offering) <br>
+     */
+    private boolean isNotPossibleToResize(VolumeVO volume, ImageFormat format, DiskOfferingVO diskOffering) {
+        ServiceOfferingJoinVO serviceOfferingView = serviceOfferingJoinDao.findById(diskOffering.getId());
+        return serviceOfferingView.getRootDiskSize() > 0 && volume.getVolumeType().equals(Volume.Type.ROOT);
```

Comment:
@Pearl1594 I changed this if-conditional when compared to PR #4829. 

I removed the ISO format comparison (`format != ImageFormat.ISO`) as the firs validation checks for `serviceOfferingView.getRootDiskSize()`. I could not see a situation of an ISO getting into this flow.

Please let me know if I am missing anything.
Thanks in advance!

Reply:
Yes @GabrielBrascher , the ISO check seems unnecessary. Looks good. Thanks

Reply:
Missed the discussion here.
@GabrielBrascher if you decide to remove ImageFormat bits, #5099 can be closed
cc @Pearl1594

Reply:
@shwstppr thanks for the ping. It seems like the case indeed. I added a commit that completely removes the format as well as some of the code used previously to retrieve the format.